src/server/pfs/server/driver_new_storage.go

```diff
@@ -132,6 +132,28 @@ func (d *driver) getFilesNewStorageLayer(ctx context.Context, repo, commit, glob
 	return tar.NewWriter(w).Close()
 }
 
+func (d *driver) listFileNS(pachClient *client.APIClient, file *pfs.File, full bool, history int64, f func(*pfs.FileInfoNewStorage) error) (retErr error) {
+	ctx := pachClient.Ctx()
+	// Validate arguments
+	if file == nil {
+		return errors.New("file cannot be nil")
+	}
+	if file.Commit == nil {
+		return errors.New("file commit cannot be nil")
+	}
+	if file.Commit.Repo == nil {
+		return errors.New("file commit repo cannot be nil")
+	}
+	// auth check
```

Comment:
This comment seems kind of unnecessary considering how descriptive the function name.